packages/client-core/src/admin/components/settings/tabs/instanceServer.tsx

```diff
@@ -53,50 +55,92 @@ const InstanceServerTab = forwardRef(({ open }: { open: boolean }, ref: React.Mu
+  const rtcEndPortValue = getSettingValue(EngineSettings.InstanceServer.RtcEndPort)
+  const rtcPortBlockSizeValue = getSettingValue(EngineSettings.InstanceServer.RtcPortBlockSize)
+  const identifierDigitsValue = getSettingValue(EngineSettings.InstanceServer.IdentifierDigits)
+  const releaseNameValue = getSettingValue(EngineSettings.InstanceServer.ReleaseName)
+  const portValue = getSettingValue(EngineSettings.InstanceServer.Port)
+  const modeValue = getSettingValue(EngineSettings.InstanceServer.Mode)
+  const locationNameValue = getSettingValue(EngineSettings.InstanceServer.LocationName)
+  const webRTCSettingsValue = getSettingValue(EngineSettings.InstanceServer.WebRTCSettings) || '{}'
 
   useEffect(() => {
-    if (instanceServerSettings) {
-      settingsState.set(instanceServerSettings)
+    if (engineSettings.length > 0) {
+      webRTCSettingsState.set(JSON.parse(webRTCSettingsValue))
       state.set({ loading: false, errorMessage: '' })
     }
-  }, [instanceServerSettings])
+  }, [engineSettings])
 
   const handleSubmit = (event) => {
     state.loading.set(true)
     event.preventDefault()
-    const newSettings = {
-      ...settingsState.get(NO_PROXY),
-      local: Boolean(settingsState.value?.local),
-      createdAt: undefined!,
-      updatedAt: undefined!
-    } as any as InstanceServerSettingType
-
-    API.instance
-      .service(instanceServerSettingPath)
-      .patch(id, newSettings)
+    const setting = {
+      WebRTCSettings: JSON.stringify(webRTCSettingsState.value),
+      local: Boolean(getSettingValue(EngineSettings.InstanceServer.Local))
+    }
+    const createOperations: Promise<EngineSettingType>[] = []
+    const updateOperations: Promise<EngineSettingType>[] = []
+
+    Object.values(EngineSettings.InstanceServer).forEach((key) => {
+      const settingInDb = engineSettings.find((el) => el.key === key)
+      if (!settingInDb) {
+        createOperations.push(
+          engineSettingMutation.create({
+            key,
+            category: 'instance-server',
+            value: setting[key],
+            type: 'private'
+          })
+        )
+      } else {
+        updateOperations.push(
+          engineSettingMutation.patch(settingInDb.id, {
+            key,
+            category: 'instance-server',
+            value: setting[key],
+            type: 'private'
+          })
+        )
+      }
+    })
+
+    Promise.all([...createOperations, ...updateOperations])
       .then(() => {
         state.set({ loading: false, errorMessage: '' })
-        instanceServerSettingQuery.refetch()
       })
       .catch((e) => {
         state.set({ loading: false, errorMessage: e.message })
       })
   }
 
   const handleCancel = () => {
-    settingsState.set(instanceServerSettings)
+    if (engineSettings.length > 0) {
+      webRTCSettingsState.set(JSON.parse(webRTCSettingsValue))
+    }
   }
 
-  if (!settingsState.value)
+  if (engineSettings.length == 0)
```

Comment:
This check should be based on `engineSettings.status` instead of length.

Reply:
![image](https://github.com/user-attachments/assets/e6cc7479-a98b-4ce3-b33d-c976a92c895a)